base_layer/wallet/src/output_manager_service/protocols/utxo_validation_protocol.rs

```diff
@@ -0,0 +1,497 @@
+            trace!(
+                target: LOG_TARGET,
+                "UTXO validation protocol (Id: {}) has no outputs to validate",
+                self.id,
+            );
+            return Ok(self.id);
+        }
+
+        let total_retries_str = match self.retry_strategy {
+            UtxoValidationRetry::Limited(n) => format!("{}", n),
+            UtxoValidationRetry::UntilSuccess => "∞".to_string(),
+        };
+
+        let mut retries = 0;
+        loop {
+            self.send_queries(outputs_to_query.clone()).await?;
+
+            let mut delay = delay_for(self.timeout).fuse();
+
+            loop {
+                futures::select! {
+                    base_node_response = base_node_response_receiver.select_next_some() => {
+                        match base_node_response {
+                            Ok(response) => if self.handle_base_node_response(response).await? {
+                            error!(target: LOG_TARGET, "Response handled with success for {} and pending_queries len: {}", self.id, self.pending_queries.len());
+                                if self.pending_queries.is_empty() {
+                                    let _ = self
+                                        .resources
+                                        .event_publisher
+                                        .send(OutputManagerEvent::UtxoValidationSuccess(self.id))
+                                        .map_err(|e| {
+                                            trace!(
+                                                target: LOG_TARGET,
+                                                "Error sending event, usually because there are no subscribers: {:?}",
+                                                e
+                                            );
+                                            e
+                                        });
+                                    return Ok(self.id);
+                                }
+                            },
+                            Err(e) => trace!(target: LOG_TARGET, "Error reading broadcast base_node_response: {:?}", e),
+                        }
+
+                    },
+                    () = delay => {
+                        break;
+                    },
+                }
+            }
+
+            info!(
+                target: LOG_TARGET,
+                "UTXO Validation protocol (Id: {}) attempt {} out of {} timed out.",
+                self.id,
+                retries + 1,
+                total_retries_str
+            );
+
+            let _ = self
+                .resources
+                .event_publisher
+                .send(OutputManagerEvent::UtxoValidationTimedOut(self.id))
+                .map_err(|e| {
+                    trace!(
+                        target: LOG_TARGET,
+                        "Error sending event, usually because there are no subscribers: {:?}",
+                        e
+                    );
```

Comment:
nit: The **only** way that the broadcast send can fail is if there are no subscribers. Also, `e` just wraps what was sent `SendError(pub T)` and (perhaps incorrectly) does not implement `Error`.

```rust
/// A **send** operation can only fail if there are no active receivers,
/// implying that the message could never be received. The error contains the
/// message being sent as a payload so it can be recovered.
```

Something like this may be more appropriate: 
```suggestion
                    trace!(
                        target: LOG_TARGET,
                        "Did not send event `{:?}` because there are no subscribers",
                        e.0
                    );
```

Reply:
True